### ISSUE TITLE
Errors with mods loaded in a different world

### ISSUE THREAD

**Issue by wapcaplet** [CleverRaven/Cataclysm-DDA#47932]
# Describe the bug

When starting a new game in a new world, the mods in *other* worlds may cause error messages.

Here is one such message:

```
 DEBUG    : Harvest list owlbear has invalid entry: owlbear_stone and adrenal_gland_large

 FUNCTION : static void harvest_list::check_consistency()
 FILE     : src/harvest.cpp
 LINE     : 149
```

Others include:
```
 DEBUG    : Harvest list owlbear_cub has invalid entry: owlbear_stone
 DEBUG    : Harvest list bulette has invalid entry: bulette_plate, bulette_pearl, and adrenal_gland_large
 DEBUG    : Harvest list demon_spider has invalid entry: demon_chitin_piece, demon_chitin_plate, and demon_spider_fang
 DEBUG    : Harvest list human_large_leather has invalid entry: adrenal_gland_large
 DEBUG    : Harvest list dragon_black has invalid entry: dragon_blood, meat_dragon, dragon_black_scale, and black_dragon_hide_raw
 DEBUG    : Harvest list lemure has invalid entry: tainted_blood
 DEBUG    : Harvest list demon_spider_queen has invalid entry: demon_chitin_piece, demon_chitin_plate, and demon_spider_fang
```

# Steps To Reproduce

1. Create a new world (A) with "Magiclysm" mod enabled (all other mods and settings at default)
2. Create a new character in the Magiclysm world (A), then save and exit
3. Create *another* new world (B) without enabling the Magiclysm mod
3. Try to create a new character in the second world (B) - get lots of errors related to Magiclysm

# Expected behavior

Mods added to world (A) should have no effect on world (B).

# Screenshots

# Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Pop; Description: Pop!_OS 20.04 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.E-9956-gf6f104a441 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm]
]

# Additional context

Found when investigating #47924

**Comment by OromisElf** [comment]
this also happens with other mods - I have a world with quite some mods and then two for testing stuff where I only have one or so mods; I didn't look terribly deep into the json errors I got but from what I can tell, having aftershock, dinomod or sweet cataclysm in a different world and not in yours also affects your current world at least to some degree

**Comment by anothersimulacrum** [comment]
Harvest lists (and possibly other things) need something like https://github.com/CleverRaven/Cataclysm-DDA/pull/33312

**Comment by PatrikLundell** [comment]
I've had it happen the other way around as well:
I saved the "real" game and created a world with one or both of the stats from kills/XP mods, existed that world and loaded the "real" game, only to see that my character had inflated stats. That happened with 0.E2 stable.

**Comment by wapcaplet** [comment]
> I saved the "real" game and created a world with one or both of the stats from kills/XP mods, existed that world and loaded the "real" game, only to see that my character had inflated stats. That happened with 0.E2 stable.

I can reproduce this in 0.E-3, but not in experimental 0.E-9956. It looks like this one was probably fixed last year by #40047.
